scrapy/itemchecker.py

```diff
@@ -0,0 +1,75 @@
+"""
+Filter for Scrapy Items to be used by Feed slots.
+"""
+from itemadapter import ItemAdapter
+
+from scrapy import Item
+from scrapy.utils.misc import load_object
+
+
+class ItemChecker:
+    """
+    This will be used by FeedExporter to decide if an item should be allowed
+    to be exported to a particular feed.
+    :param feed_options: feed specific options passed from FeedExporter
+    :type feed_options: dict
+    """
+
+    item_classes = ()
+
+    def __init__(self, feed_options):
+        self.feed_options = feed_options
+
+        if 'item_classes' in self.feed_options:
+            item_classes = self.feed_options['item_classes']
+
+            if isinstance(item_classes, (Item, str)):
```

Comment:
I would suggest requiring the value to be a list. It will not be much of a trouble for users, it will keep their code consistent (`'item_classes': class` seems semantically wrong, `'item_classes': [class]` seems more consistent with the option name), and it will simplify your implementation (you don’t need an `if` statement here at all).

Internally, because you are merging the user-input option with a class-defined variable, you may want to enforce the class-defined variable to be a set, and convert the user-input option into a set before you merge them. Or you could remove the class-defined variable, and only filter item classes defined by users.